src/ai/defaultai.cc

```diff
@@ -2882,10 +3073,9 @@ bool DefaultAI::construct_building(const Time& gametime) {
 					// Overdue priority here
 					prio += bo.primary_priority;
 
-					prio += bf->supporters_nearby.at(bo.get_collected_map_resource()) * 5;
+					prio += number_of_supporters_nearby * 5;
 
-					prio += (bf->critters_nearby * 3) - 8 -
-					        5 * bf->collecting_producers_nearby.at(bo.get_collected_map_resource());
+					prio += (bf->critters_nearby * 3) - 8 - 5 * number_of_supported_producers_nearby;
```

Comment:
another wrong usage of number_of_supported_producers_nearby here we should have identified other hunters